packages/alfa-url/src/url.ts

```diff
@@ -354,11 +400,33 @@ export namespace URL {
           // "#" which we need to remove.
           Option.of(hash)
             .reject(isEmpty)
-            .map((hash) => hash.replace(/^#/, ""))
+            .map((hash) => hash.replace(/^#/, "")),
+
+          // The URL cannot be used as a base URL when the scheme either isn't
+          // special or the pathname doesn't start with a leading "/".
+          !(isSpecialScheme(scheme) || pathname[0] === "/")
```

Comment:
Comment and code don't match.
Code test "scheme not special **and** path doesn't start with /" but comment says **or**.
Given that tests are working, I guess that code is correct. Thus said, I also suggest to De Morgan-ise it to better match the comment.
```suggestion
          // The URL cannot be used as a base URL when the scheme isn't
          // special and the pathname doesn't start with a leading "/".
          !isSpecialScheme(scheme) && pathname[0] !== "/"
```

Reply:
Good catch!